controllers/Backend.php

```diff
@@ -1,104 +1,344 @@
-        $model_dashboard = new Dashboard();
-        $model_user = new User();
+        $manager_comment = new CommentManager();
+        $comments = $manager_comment->get_comments(); 
+       
+        if(!isset($_SESSION['admin'])){
+            header("location: /articles-a-la-une");
+        }
+        
+        if(filter_has_var(INPUT_GET, 'delete')){
+            $comment = $manager_comment->delete_comment();
+        }
+                
+        if(filter_has_var(INPUT_GET, 'approve')){
+            $comment = $manager_comment->approve_comment();
+        }
         
-        $page="backend/dashboard";
-        $topbar="backend/topbar";
-        require "views/layout.php";
+        require_once "config/dashboard.php";
+        $myView = new View('dashboard', 'backend');
+        $myView->render(array('comments' => $comments));
     }
 
     public function write()
     {
-        $model_article = new Article();
-        $model_user = new User();
+        $manager_article = new ArticleManager();
+        
+        //Permet de vérifier les données envoyées par l'utilsiateur
+        if(filter_has_var(INPUT_POST, 'post')){
+            if(filter_has_var(INPUT_POST, 'title')){
+                $title = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'title')));
+            }
+            if(filter_has_var(INPUT_POST, 'chapo')){
+                $chapo = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'chapo')));
+            }
+           if(filter_has_var(INPUT_POST, 'content')){
+                $content = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'content')));
+            }
+
+            $posted = filter_input(INPUT_POST, 'public') ? "1" : "0";
+
+            $errors = [];
+
+            if(empty($title) || empty($content) || empty($chapo)){
+                $errors['empty'] = "Veuillez remplir tous les champs";
+            }
+            if(strlen($title) < 5){
+                $errors['title'] = "Votre titre doit contenir au moins 5 caractères.";
+            }
+            if(strlen($chapo) < 5){
+                $errors['chapo'] = "Votre chapô doit contenir au moins 5 caractères.";
+            }
+            if(strlen($title) < 5){
+                $errors['content'] = "Votre article doit contenir au moins 100 caractères.";
+            }
+
+            if(!empty($_FILES['image']['name'])){
+                $file = $_FILES['image']['name'];
+                $extensions = ['.png','.jpg','.jpeg','.gif','.PNG','.JPG','.JPEG','.GIF'];
+                $extension = strrchr($file,'.');
+                if(!in_array($extension,$extensions)){
+                    $errors['image'] = "Cette image n'est pas valable";
+                }
+            }
+
+            if(empty($errors)){
+                $manager_article->post($title,$chapo,$content,$posted);
+                if(!empty($_FILES['image']['name'])){
+                    $manager_article->post_img($_FILES['image']['tmp_name'], $extension);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Direct use of $_FILES Superglobal detected.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)